src/pip/_internal/operations/prepare.py

```diff
@@ -106,15 +104,19 @@ def unpack_vcs_link(link, location):
 
 
 class File(object):
+
     def __init__(self, path, content_type):
-        # type: (str, str) -> None
+        # type: (str, Optional[str]) -> None
         self.path = path
-        self.content_type = content_type
+        if content_type is None:
+            self.content_type = mimetypes.guess_type(path)[0]
+        else:
+            self.content_type = content_type
 
 
 def get_http_url(
     link,  # type: Link
-    downloader,  # type: Downloader
+    download,  # type: Downloader
```

Comment:
I definitely prefer "downloader" as the name for the variable holding `Downloader` instances. :)

Reply:
This is how I interpret your review :laughing:

![mixed-signal](https://user-images.githubusercontent.com/13689192/90120165-a6ad0580-dd84-11ea-9775-3b54bab46b2f.jpg)

Jokes aside I think it'd be nicer for you to read `Downloader` as `Callable[..., ...]`.  TBH I've been pushing my agenda on eliminating the `downloader(...)` call for quite a few PRs and I really hope you and other maintainers of pip can put up with it.

Reply:
Naming convention around this kind of first-class callables is an actively debated topic, unfortunately. Maybe a better way to do this would be to rename the `Downloader` type to something that expresses the fact the instance is a callable, not an arbitrary object.

Reply:
> This is how I interpret your review 😆

Haha, it's more of "let's ship it, and here's a non-blocking concern". :)